src/basics.md

```diff
@@ -326,6 +327,80 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-parse-subprocess-input]: #ex-parse-subprocess-input
+<a name="ex-parse-subprocess-input"></a>
+## Run an external command passing it stdin and check for an error code
+
+[![regex-badge]][regex] [![cat-os-badge]][cat-os] [![cat-text-processing-badge]][cat-text-processing]
+
+Opens the `python` interpreter using an external [`Command`] and passes it a python statement
+for execution. [`Output`] of said statement is then parsed using [`Regex`] and aggregated for all unique words used in output.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+use regex::Regex;
+use std::io::Write;
+use std::ops::Deref;
+use std::process::{Command, Stdio};
+
+# error_chain!{
+#     foreign_links {
+#         Io(std::io::Error);
+#         Regex(regex::Error);
+#         Utf8(std::string::FromUtf8Error);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let mut child = Command::new("python").stdin(Stdio::piped())
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .spawn()?;
+
+    {
+        match child.stdin.as_mut() {
+            None => return Err("Child process stdin has not been captured!".into()),
+            Some(stdin) => {
+                stdin.write_all(b"import this; copyright(); credits(); exit()")?;
+            }
+        }
+    }
+
+    let output = child.wait_with_output()?;
+
+    if output.status.success() {
+        match String::from_utf8(output.stdout) {
+            Err(e) => Err(e.into()),
+            Ok(s) => {
+                let re = Regex::new(r"[[:punct:]&&[^']]")?;
+                {
+                    let stripped = re.replace_all(&s, " ");
+                    let mut result = stripped.deref()
+                        .split_whitespace()
+                        .map(|s| s.to_lowercase())
+                        .collect::<Vec<String>>();
+                    result.sort();
+                    result.dedup();
+                    println!("Found {} unique words:", result.len());
+                    println!("{:#?}", result);
+                }
+                Ok(())
+            }
+        }
+    } else {
+        match String::from_utf8(output.stderr) {
```

Comment:
Instead of this whole match you can just use `bail!` here. It even takes care of formatting